daemon/src/wallet.rs

```diff
@@ -40,6 +42,7 @@ pub struct Actor {
     used_utxos: HashSet<OutPoint>,
```

Comment:
You need to clear these upon re-init for your implementation to be correct.

(Side note: If we would restart the entire actor, this would be automatically handled. Managing state is hard :upside_down_face: )